test_erezutils.py

```diff
@@ -0,0 +1,97 @@
+import hashlib
+import tempfile
+import unittest
+
+from erezutils import chunks, hashfiles, rupdate
+
+
+class RecursiveUpdateTest(unittest.TestCase):
+    def test_update_empty(self):
+        self.assertEqual({"a": 1}, rupdate({}, {"a": 1}))
+
+    def test_update_existing(self):
+        self.assertEqual({"a": 2}, rupdate({"a": 1}, {"a": 2}))
+
+    def test_update_with_other_keys(self):
+        self.assertEqual({"a": 2, "b": 1}, rupdate({"a": 1}, {"a": 2, "b": 1}))
+
+    def test_update_leaves_other_keys_untouched(self):
+        self.assertEqual({"a": 2, "b": 1}, rupdate({"a": 1, "b": 1}, {"a": 2}))
+
+    def test_recursive_update(self):
+        self.assertEqual({
+            "a": {"b": 2}},
+            rupdate({"a": {"b": 1}}, {"a": {"b": 2}})
+        )
+
+    def test_recursive_update_on_empty_dict(self):
+        self.assertEqual({"a": {"b": 2}}, rupdate({}, {"a": {"b": 2}}))
+
+    def test_recursive_update_with_other_keys(self):
+        self.assertEqual(
+            {"a": {"b": 2, "c": 1}},
+            rupdate({"a": {"b": 1}}, {"a": {"b": 2, "c": 1}})
+        )
+
+    def test_recursive_update_leaves_other_keys_untouched(self):
+        self.assertEqual(
+            {"a": {"b": 2, "c": 1}},
+            rupdate({"a": {"b": 1, "c": 1}}, {"a": {"b": 2}})
+        )
+
+
+class HashfilesTest(unittest.TestCase):
+    def test_hash_default_sha512(self):
+        data = b"thisisatest"
+        data_sha512 = hashlib.sha512()
+        data_sha512.update(data)
+
+        with tempfile.TemporaryFile(mode="w+b") as f:
```

Comment:
WDYT about using `io.BytesIO` to avoid unnecessary filesystem writes during tests?